codecs/src/aper/encode/mod.rs

```diff
@@ -202,6 +202,24 @@ pub fn encode_length_determinent(
 ) -> Result<(), AperCodecError> {
     log::trace!("encode_length_determinent");
 
+    if let Some(l) = lb {
+        if value < l as usize {
+            return Err(AperCodecError::new(format!(
+                "Cannot encode length determinent {} - less than lower bound {}",
+                value, l,
+            )));
+        }
+    }
+
+    if let Some(u) = ub {
+        if value > u as usize {
+            return Err(AperCodecError::new(format!(
+                "Cannot encode length determinent {} - greater than upper bound {}",
+                value, u,
+            )));
+        }
+    }
+
```

Comment:
This code is not required, because this again calls `encode_constrained_whole_number` and it can be checked there. If at all you want to return appropriate error for debugging, one can `match` on the return value of the `encode_constrained_whole_number` and add appropriate error message (like `map_err` you've described somewhere else). This way this check can be avoided.

Reply:
@nplrkn : Can you address this comment please? So that we can close this PR?

Reply:
@nplrkn : I am looking at resolution of this comment. That is either I am missing something about why you should be having this check in this function or you may want to drop this change completely and the one only inside the `encode_constrained_whole_number` is sufficient. Once done, I will merge this.

Reply:
In addition, I believe - a `rebase` is required for this PR as it's quite old and a lot has changed.

Reply:
@nplrkn : I think this comment is still open. Do you plan to fix this?